roc/log_test.go

```diff
@@ -180,3 +181,51 @@ func TestLog_Levels(t *testing.T) {
 		})
 	}
 }
+
+func TestLog_Write(t *testing.T) {
+	SetLogLevel(LogDebug)
+	defer SetLogLevel(defaultLogLevel)
+
+	ch := make(chan LogMessage, 1)
+
+	SetLoggerFunc(func(msg LogMessage) {
+		if msg.Level == LogDebug &&
+			msg.Module == "roc_go" &&
+			strings.Contains(msg.Text, "entering OpenContext()") {
+			select {
+			case ch <- msg:
+			default:
+			}
+		}
+	})
+	defer SetLoggerFunc(nil)
+
+	testStartTime := time.Now()
+	ctx, err := OpenContext(ContextConfig{})
+	require.NoError(t, err)
+
+	select {
+	case msg := <-ch:
+		assert.Equal(t, LogDebug, msg.Level, "Expected log level to be debug")
+		assert.Equal(t, "roc_go", msg.Module)
+		assert.Equal(t, "roc/context.go", msg.File)
+		assert.Greater(t, msg.Line, 0, "Line number must be positive")
+		assert.Equal(t, uint64(os.Getpid()), msg.Pid)
+		assert.NotEmpty(t, msg.Tid)
+		assert.True(t,
+			msg.Time.After(testStartTime.Add(-time.Millisecond)) &&
+				msg.Time.Before(testStartTime.Add(time.Millisecond)),
+			"Time assertion failed: test time is not within the tolerance of the start time of the test",
+		)
+		assert.True(t,
+			msg.Time.After(time.Now().Add(-time.Millisecond)) &&
+				msg.Time.Before(time.Now().Add(time.Millisecond)),
+			"Time assertion failed: message time cannot be greater than the time now",
+		)
```

Comment:
```suggestion
		assert.WithinRange(t, msg.Time, testStartTime, time.Now(),
			"Time must have meaningful value")
```

Reply:
This way there is no need for magic millisecond because the range is inclusive. Also the code is much shorter.

Reply:
WithinRange doesn't seem to exist on the assert package

Reply:
assert.WithinRange doesn't exist though